### DISCUSSION TITLE
Template Command Checks Against Future Version of Kubernetes

### DISCUSSION THREAD

**Discussion by kylegoch** [helmfile/helmfile#181] [Questions]
Kubernetes: 1.21.12
Helm: 3.9.0
helm-diff: 3.5.0
helmfile: 0.144.0

When I run `helmfile template` i receive this warning on one of my charts:
```
STDERR:
  Error: chart requires kubeVersion: < 1.23.0-0 which is incompatible with Kubernetes v1.24.0
  Use --debug flag to render out invalid YAML
```

The chart in question really does have a lock of <1.23 and thats fine as I am on 1.21. `diff` and `apply` work just fine for the helmfile and chart in question. I can also get `helmfile template` to work by adding `--validate`.

However it seems the spirit of `template` is that you can render the charts out without needing kubernetes access. But adding `--validate` breaks that thinking.

Is there any way to have `helmfile template` render for a specified kubernetes version? It seems to be just be defaulting to latest (1.24) which is fine, but would love to be able to point at different versions.

**Comment by yxxhero** [comment]
@kylegoch I wiil see this.

**Reply by yxxhero** [reply]
```
# DEPRECATED: This is available only on Helm 2, which has been EOL since 2020
# Configure a Kubernetes version to  pass to 'helm template' via the --kube-version flag:
# See https://github.com/roboll/helmfile/pull/2002 for more information.
kubeVersion: v1.21
```
